src/libraries/Common/src/System/Security/Cryptography/RSASecurityTransforms.cs

```diff
@@ -411,14 +412,25 @@ public override bool TryDecrypt(ReadOnlySpan<byte> data, Span<byte> destination,
                     throw new CryptographicException(SR.Cryptography_RSA_DecryptWrongSize);
                 }
 
-                if (padding.Mode == RSAEncryptionPaddingMode.Pkcs1 ||
+                if ((padding.Mode == RSAEncryptionPaddingMode.Pkcs1 && HasWorkingPKCS1Padding) ||
                     padding == RSAEncryptionPadding.OaepSHA1)
                 {
                     return Interop.AppleCrypto.TryRsaDecrypt(privateKey, data, destination, padding, out bytesWritten);
```

Comment:
I thought I remembered seeing OAEP+hash algorithm switch statements from a previous PR... I thought we were done doing manual OAEP padding on macOS?

Reply:
I didn't switch the managed code to use it. I can submit that as a small separate PR.